### PR TITLE
Fix for node contributors permissions query filter

### PR DESCRIPTION
**Pull request by alexschiller**
# [OSF-4737]
## Purpose
Query filter for node contributors now correctly accepts 'permissions' and does not accept 'permission'

## Changes 
rename variable in nodes serializer to permissions (from permission) and add source = 'permission'

## Side Effects
None

## Testing notes
The same testing procedure used for https://github.com/CenterForOpenScience/osf.io/pull/4359 would be useful here

### CODE REVIEW

**Comment by sloria** [comment]
@alexschiller Can you please look into the failed tests?

**Comment by abought** [comment]
The Travis reporting indicator here is... interesting.
screen shot 2015-10-26 at 3 09 00 pm

When run locally, 8 tests fail due to missing keys in output data, undefined variables, etc. Please fix the tests, and I'll ping Steve to look into Travis' generous definition of success.

File: api_tests/nodes/views/test_node_contributors_list.py [hunk 1]
```diff
@@ -159,6 +159,16 @@ def test_filtering_on_invalid_field(self):
         errors = res.json['errors']
         assert_equal(len(errors), 1)
         assert_equal(errors[0]['detail'], 'Query string contains an invalid filter.')
+    
+    def test_filtering_on_permission_field(self):
+        # regression test for changes in filter fields
+        # permission is now permissions
+        url_permission = '/{}nodes/{}/contributors/?filter[permission]=foo'.format(API_BASE, self.project._id)
+        res = self.app.get(url_fullname, auth=self.user.auth, expect_errors=True)
```
Comment (abought) [inline api_tests/nodes/views/test_node_contributors_list.py]: This variable is not defined in scope, and the test fails when run.

File: api_tests/nodes/views/test_node_contributors_list.py [hunk 2]
```diff
@@ -159,6 +159,16 @@ def test_filtering_on_invalid_field(self):
         errors = res.json['errors']
         assert_equal(len(errors), 1)
         assert_equal(errors[0]['detail'], 'Query string contains an invalid filter.')
+    
+    def test_filtering_on_permission_field(self):
+        # regression test for changes in filter fields
+        # permission is now permissions
+        url_permission = '/{}nodes/{}/contributors/?filter[permission]=foo'.format(API_BASE, self.project._id)
```
Comment (abought) [inline api_tests/nodes/views/test_node_contributors_list.py]: Should this now be `filter[permissions]`?

Comment (abought) [inline api_tests/nodes/views/test_node_contributors_list.py]: Ah, filtering the valid field on the unknown value just returns no results, but a success error code. Ok.

**Comment by abought** [comment]
CR1 pass complete, but failing tests need to be addressed (and reviewed) before this can go forward.

**Comment by sloria** [comment]
Closing; I believe this was fixed in the latest updates to the contributor page.

**Pull request closed without merging**